tests/sentry/api/endpoints/test_organization_dashboards.py

```diff
@@ -418,6 +418,108 @@ def test_get_exclude_favorites_with_sort(self):
             values = [row["title"] for row in response.data]
             assert values == ["General", "Dashboard 1", "Dashboard 2", "Dashboard 7", "Dashboard 6"]
 
+    def test_pin_favorites_with_my_dashboards_sort(self):
+        user_1 = self.create_user(username="user_1")
+        self.create_member(organization=self.organization, user=user_1)
+
+        Dashboard.objects.create(
+            title="Dashboard A",
+            created_by_id=self.user.id,
+            organization=self.organization,
+        )
+        dashboard_B = Dashboard.objects.create(
+            title="Dashboard B",
+            created_by_id=user_1.id,
+            organization=self.organization,
+        )
+        dashboard_C = Dashboard.objects.create(
+            title="Dashboard C",
+            created_by_id=user_1.id,
+            organization=self.organization,
+        )
+        dashboard_D = Dashboard.objects.create(
+            title="Dashboard D",
+            created_by_id=self.user.id,
+            organization=self.organization,
+        )
+        dashboard_E = Dashboard.objects.create(
+            title="Dashboard E",
+            created_by_id=user_1.id,
+            organization=self.organization,
+        )
+
+        dashboard_B.favorited_by = [self.user.id]
+        dashboard_D.favorited_by = [self.user.id]
+        dashboard_E.favorited_by = [self.user.id]
+        dashboard_C.favorited_by = [user_1.id]
+
+        with self.feature("organizations:dashboards-favourite"):
+            response = self.client.get(self.url, data={"sort": "mydashboards", "pin": "favorites"})
```

Comment:
not related to this PR and the favourite feature, but this makes me think that we should change `mydashboards` into a pin one day!